deploy/examples/create-external-cluster-resources.py

```diff
@@ -523,40 +516,24 @@ def create_cephCSIKeyring_cephFSProvisioner(self):
             cephfs_filesystem = self._arg_parser.cephfs_filesystem_name
             if cephfs_filesystem == "":
                 entity = "client.csi-cephfs-provisioner-{}".format(cluster_name)
-                cmd_json = {"prefix": "auth get-or-create",
-                            "entity": entity,
-                            "caps": ["mon", "allow r, allow command 'osd blocklist'", 
-                                     "mgr", "allow rw",
-                                     "osd", "allow rw tag cephfs metadata=*"],
-                            "format": "json"}
+                caps = {"mon": "allow r, allow command 'osd blocklist'", 
+                        "mgr": "allow rw",
+                        "osd": "allow rw tag cephfs metadata=*"}
             else:
                 entity = "client.csi-cephfs-provisioner-{}-{}".format(cluster_name,cephfs_filesystem)
-                cmd_json = {"prefix": "auth get-or-create",
-                            "entity": entity,
-                            "caps": ["mon", "allow r, allow command 'osd blocklist'",
-                                     "mgr", "allow rw",
-                                     "osd", "allow rw tag cephfs metadata={}".format(cephfs_filesystem)],
-                            "format": "json"}
+                caps = {"mon": "allow r, allow command 'osd blocklist'",
+                        "mgr": "allow rw",
+                        "osd": "allow rw tag cephfs metadata={}".format(cephfs_filesystem)}
         else:
-            cmd_json = {"prefix": "auth get-or-create",
-                        "entity": entity,
-                        "caps": ["mon", "allow r, allow command 'osd blocklist'",
-                                 "mgr", "allow rw",
-                                 "osd", "allow rw tag cephfs metadata=*"],
-                        "format": "json"}
-        if self._arg_parser.dry_run:
-            return self.dry_run("ceph " + cmd_json['prefix'] + " " + cmd_json['entity'] + " " + " ".join(cmd_json['caps']))
-        ret_val, json_out, err_msg = self._common_cmd_json_gen(cmd_json)
-        # if there is an unsuccessful attempt,
-        if ret_val != 0 or len(json_out) == 0:
-            raise ExecutionFailureException(
-                "'auth get-or-create client.csi-cephfs-provisioner' command failed.\n" +
-                "Error: {}".format(err_msg if ret_val != 0 else self.EMPTY_OUTPUT_LIST))
-        return str(json_out[0]['key'])
-
-    def create_cephCSIKeyring_cephFSNode(self):
+            caps = {"mon": "allow r, allow command 'osd blocklist'",
+                    "mgr": "allow rw",
+                    "osd": "allow rw tag cephfs metadata=*"}
+        
+        return caps,entity
+    
+    def get_cephfs_node_caps_and_entity(self):
```

Comment:
Similarly this function as well,

```suggestion
    def get_cephfs_node_caps_and_entity(self):
        entity = "client.csi-cephfs-node"
        caps = {"mon": "allow r, allow command 'osd blocklist'",
                "mgr": "allow rw",
                "osd": "allow rw tag cephfs *=*",
                "mds": "allow rw"}
        if self._arg_parser.restricted_auth_permission:
            cluster_name = self._arg_parser.cluster_name
            if cluster_name == "":
                raise ExecutionFailureException(
                    "cluster_name not found, please set the '--cluster-name' flag")
            cephfs_filesystem = self._arg_parser.cephfs_filesystem_name
            if cephfs_filesystem == "":
                entity = "{}-{}".format(entity, cluster_name)
            else:
                entity = "{}-{}-{}".format(entity, cluster_name, cephfs_filesystem)
                caps["osd"] = "allow rw tag cephfs data={}".format(cephfs_filesystem)
            
        return caps,entity    
```